node/router/tests/disconnect.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+mod common;
+use common::*;
+
+use snarkos_node_tcp::{protocols::Handshake, P2P};
+
+use core::time::Duration;
+
+macro_rules! print_tcp {
```

Comment:
nit: since this macro is duplicated (macro imports can be annoying), I'd recommend moving it to the `common` module for the tests

Reply:
Agreed, I have deduped the test macro into `common`